sphinx_gherkindoc/pytest_bdd_writer.py

```diff
@@ -0,0 +1,284 @@
+                else:
+                    continue
+            step_glossary[step.name.lower()].add_reference(
+                step.name,
+                pathlib.Path(_step_filename(step)).resolve().relative_to(root_path),
+                _step_line(step),
+            )
+            formatted_step = format_step(step, this_step_format)
+            output_file.add_output(f"| {formatted_step}")
+            step_table = _step_table(step)
+            if step_table:
+                output_file.blank_line()
+                table(step_table, inline=True)
+                output_file.blank_line()
+            else:
+                step_text = _step_text(step)
+                if step_text:
+                    text(step_text)
+
+    def examples(
+        scenario: Optional[pytest_bdd.feature.Scenario],
+        feature: pytest_bdd.feature.Feature,
+    ) -> None:
+        if scenario:
+            obj = scenario
+            tag_objects = [scenario, feature]
+        else:
+            obj = feature
+            tag_objects = []
+        example_obj = getattr(obj, "examples", [])
+        if not example_obj.examples:
+            return
+        section(3, example_obj)
+        tags(obj.tags, *tag_objects)
+        table(example_obj)
+        output_file.blank_line()
+
+    def table(table: PytestTable, inline: bool = False) -> None:
+        indent_by = INDENT_DEPTH if inline else 0
+        directive = ".. csv-table::"
+        output_file.add_output(directive, indent_by=indent_by)
+        headers = '", "'.join(table.example_params)
+        indent_by += INDENT_DEPTH
+        output_file.add_output(f':header: "{headers}"', indent_by=indent_by)
+        output_file.add_output(f":quote: {QUOTE}", line_breaks=2, indent_by=indent_by)
+        for row in table.examples:
+            row = f"{QUOTE}, {QUOTE}".join(map(rst_escape, row))
+            output_file.add_output(f"{QUOTE}{row}{QUOTE}", indent_by=indent_by)
+
+    # Declare roles in each rST file
+    # so that they are available for customization by users.
+    for role in AVAILABLE_ROLES:
+        output_file.add_output(f".. role:: {role}")
+    if AVAILABLE_ROLES:
+        output_file.blank_line()
+
+    feature = pytest_bdd.feature.Feature(
+        root_path, pathlib.Path(source_path).resolve().relative_to(root_path)
+    )
+    section(1, feature)
+    description(feature)
+    if feature.examples.examples:
+        examples(None, feature)
+    if feature.background and not integrate_background:
+        section(2, feature.background)
+        steps(feature.background.steps, is_feature_background=True)
+        output_file.blank_line()
+    for scenario in feature.scenarios.values():
+        section(2, scenario)
+        tags(scenario.tags, feature)
+        description(scenario)
+        steps(scenario.steps, integrate_background=integrate_background)
+        output_file.blank_line()
+        examples(scenario, feature)
+
+    return output_file
```

Comment:
Ugh, I can't tell just how duplicated this is using the GIthub PR review UI alone.
I'll have to pull the PR and do some diffing and what not.
(So perhaps not tonight unfortunately, hopefully tomorrow)